test/e2e/test_image.py

```diff
@@ -95,3 +96,177 @@ def test_create_with_invalid_image_file(self):
+                "entity_id": entity_id_a,
+                "id": image_b_id,
+                "title": "Essay Title",
+                "description": "A research essay.",
+            },
+            {
+                **IMAGE_GET_DATA_ALL_VALUES,
+                "entity_id": entity_id_b,
+                "id": image_c_id,
+                "title": "Book Title",
+                "description": "A science book.",
+            },
+        ]
+
+
+class TestList(ListDSL):
+    """Tests for getting a list of images."""
+
+    def test_list_with_no_filters(self):
+        """
+        Test getting a list of all images with no filters provided.
+
+        Posts 3 images and expects it to be returned.
+        """
+
+        expected_result = self.post_test_images()
+        self.get_images()
+        self.check_get_images_success(expected_result)
+
+    def test_list_with_entity_id_filter(self):
+        """
+        Test getting a list of all images with an `entity_id` filter provided.
+
+        Posts 3 images and then filter using the `entity_id`.
+        """
+
+        expected_result = self.post_test_images()[:2]
+        self.get_images(filters={"entity_id": expected_result[0]["entity_id"]})
+        self.check_get_images_success(expected_result)
+
+    def test_list_with_entity_id_filter_with_no_matching_results(self):
+        """
+        Test getting a list of all images with an `entity_id` filter provided.
+
+        Posts 3 images and expects no results.
+        """
+        self.post_test_images()
+        self.get_images(filters={"entity_id": ObjectId()})
+        self.check_get_images_success([])
+
+    def test_list_with_primary_filter(self):
+        """
+        Test getting a list of all images with a `primary` filter provided.
+
+        Posts 3 images and then filter using the `primary`.
+        """
+
+        expected_result = self.post_test_images()
+        self.get_images(filters={"primary": False})
+        self.check_get_images_success(expected_result)
+
+    def test_list_with_primary_filter_with_no_matching_results(self):
+        """
+        Test getting a list of all images with a `primary` filter provided.
+
+        Posts 3 images and expects no results.
+        """
+        self.post_test_images()
+        self.get_images(filters={"primary": True})
+        self.check_get_images_success([])
+
+    def test_list_with_invalid_primary_filter(self):
+        """
+        Test getting a list of all images with an invalid `primary` filter provided.
+
+        Posts 3 images and expects a 422 status code.
+        """
+        self.post_test_images()
+        self.get_images(filters={"primary": str(ObjectId())})
+        self.check_images_list_response_failed_with_message(
+            422,
+            "Input should be a valid boolean, unable to interpret input",
+            self._get_response_image.json()["detail"][0]["msg"],
+        )
+
+    def test_list_with_invalid_entity_id_filter(self):
```

Comment:
Could this test be moved to be at the end of the entity_id tests so they are grouped.